src/WebJobs.Script.Grpc/Channel/GrpcCapabilities.cs

```diff
@@ -26,18 +26,34 @@ public string GetCapabilityState(string capability)
             return null;
         }
 
-        public void UpdateCapabilities(IDictionary<string, string> capabilities)
+        public void UpdateCapabilities(IDictionary<string, string> capabilities, GrpcCapabilitiesUpdateStrategy strategy)
         {
             if (capabilities == null)
             {
                 return;
             }
 
-            _logger.LogDebug($"Updating capabilities: {capabilities.ToString()}");
+            if (strategy == GrpcCapabilitiesUpdateStrategy.Merge)
+            {
+                _logger.LogDebug($"Updating capabilities using merge strategy: {capabilities.ToString()}");
 
-            foreach (KeyValuePair<string, string> capability in capabilities)
+                foreach (KeyValuePair<string, string> capability in capabilities)
+                {
+                    UpdateCapability(capability);
+                }
+            }
+            else if (strategy == GrpcCapabilitiesUpdateStrategy.Replace)
+            {
+                _logger.LogDebug($"Updating capabilities using replace strategy: {capabilities.ToString()}");
+                _capabilities.Clear();
+                foreach (KeyValuePair<string, string> capability in capabilities)
+                {
+                    UpdateCapability(capability);
+                }
+            }
+            else
             {
-                UpdateCapability(capability);
+                throw new InvalidOperationException($"Did not recognize the capability update strategy {nameof(strategy)}.");
```

Comment:
You don't want `{nameof(strategy)}` here -- that'll write out "strategy". You want `{strategy}` which will write out the enum value.